### PR TITLE
Add `Clash.Cores.Xilinx.Unisim.DnaPortE2`

### PR DESCRIPTION
**Pull request by martijnbastiaan**
## Still TODO:

  - [X] ~~Write a changelog entry (see changelog/README.md)~~
  - [X] Check copyright notices are up to date in edited files. Note: upstreamed from Bittide project, so license change.

### CODE REVIEW

File: clash-cores/src/Clash/Cores/Xilinx/Unisim/DnaPortE2/Internal.hs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  | Shift (BitVector 96)
+  -- ^ Shift out a bit from the DNA register
+  | Done (BitVector 96)
+  -- ^ Valid DNA shifted out
+  deriving (Generic, NFDataX)
+
+type DIN = Bit
+type DOUT = Bit
+type READ = Bool
+type SHIFT = Bool
```
Comment (DigitalBrains1) [inline clash-cores/src/Clash/Cores/Xilinx/Unisim/DnaPortE2/Internal.hs:26]: Why is it all caps?

**Review by DigitalBrains1 — COMMENTED** [review]
I did not read the relevant datasheet, but LGTM!

File: clash-cores/src/Clash/Cores/Xilinx/Unisim/DnaPortE2/Internal.hs (hunk 1 above)
Comment (martijnbastiaan) [reply clash-cores/src/Clash/Cores/Xilinx/Unisim/DnaPortE2/Internal.hs:26]: Probably because it was in the datasheet. Will change!

**Comment by martijnbastiaan** [comment]
> I did not read the relevant datasheet, but LGTM!

Great! We do use it "in production" already. Given the simulation test, and the simplicity of the `inst` implementation, I feel pretty confident this is okay! :)

**Pull request merged**